### PR TITLE
Rework == operator on Rect to take standardization into account

### PR DESCRIPTION
**Pull request by rjpilgrim**
A simple test showing the operator at work is also included..

### CODE REVIEW

File: Tests/CoreGraphicsTests/CoreGraphicsTests.swift [hunk 1]
```diff
@@ -203,6 +203,13 @@ final class CoreGraphicsTests: XCTestCase {
     XCTAssertFalse(r4.intersects(r1))
   }
 
+  func testRectNonstandardEquality() {
+    let rectOne: Rect = Rect(x: 0, y: 0, width: 10, height: 10)
+    let rectTwo: Rect = Rect(x: 10, y: 10, width: -10, height: -10)
+
+    XCTAssertTrue(rectOne == rectTwo)
+  }
```
Comment (compnerd) [inline Tests/CoreGraphicsTests/CoreGraphicsTests.swift:211]: Better test coverage for this would be nice, but that can be a follow up.

**Review by compnerd — APPROVED** [review]

**Comment by compnerd** [comment]
Interesting, this seems to be causing a crash somehow?

File: Sources/SwiftWin32/CG/Rect.swift [hunk 2]
```diff
@@ -211,6 +211,12 @@ public struct Rect {
 }
 
 extension Rect: Equatable {
+  // MARK - Operator Functions
+  public static func == (lhs: Rect, rhs: Rect) -> Bool {
+    guard !lhs.isNull, !rhs.isNull else { return false }
```
Comment (compnerd) [inline Sources/SwiftWin32/CG/Rect.swift:216]: Actually, this seems wrong .. the null rect should be equal no?

Comment (rjpilgrim) [reply Sources/SwiftWin32/CG/Rect.swift:216]: Good point, that line can just go away.

**Comment by rjpilgrim** [comment]
> Interesting, this seems to be causing a crash somehow?

Yes, but I do not understand it. Commenting out the entire new function for `==` causes everything to work again, with the my new test failing as expected.

**Comment by compnerd** [comment]
Its unclear why it is failing, would require running under a debugger.

**Comment by rjpilgrim** [comment]
> ~Its unclear why it is failing, would require running under a debugger.~ Its infinite recursion, there is a stack overflow (`isNull` invokes `==`, invokes `isNull`, repeat ad-nauseum)

And Standardized calls isNull... I think there is need for a private non-standardized equality function just to implement isNull.

**Comment by compnerd** [comment]
> > ~Its unclear why it is failing, would require running under a debugger.~ Its infinite recursion, there is a stack overflow (`isNull` invokes `==`, invokes `isNull`, repeat ad-nauseum)
> 
> And Standardized calls isNull... I think there is need for a private non-standardized equality function just to implement isNull.

Sure, I don't have an idea off the top of my head how to break the cycle, so restructuring with a `@inline(__always) private` function should be fine.

**Comment by rjpilgrim** [comment]
Test is now running correctly.

**Pull request merged**